hazelcast/src/main/java/com/hazelcast/config/security/SimpleAuthenticationConfig.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config.security;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.hazelcast.config.LoginModuleConfig;
+import com.hazelcast.config.LoginModuleConfig.LoginModuleUsage;
+
+/**
+ * Typed authentication configuration for {@code SimplePropertiesLoginModule}. The user configuration (username, password, and
+ * assigned roles) is directly part of the configuration object.
+ */
+public class SimpleAuthenticationConfig extends AbstractClusterLoginConfig<SimpleAuthenticationConfig> {
+
+    private final Map<String, UserDto> userMap = new ConcurrentHashMap<>();
+    private volatile String roleSeparator;
```

Comment:
I would keep it, but don't promote it. I.e. it wouldn't be used in code samples or tutorials. It would live as a note in the reference manual.

I use volatile because sometimes the security config is used from other threads than the one used for the node initialization.